### ISSUE TITLE
"Blazor.*" namespace causes a lot of compatibility friction

### ISSUE THREAD

**Issue by Xeevis** [ljbc1994/BlazorIntersectionObserver#26]
## Describe the bug
Due to the way namespaces are resolved, It's impossible to reference any components from this library in any project that contains `*.Blazor.*` anywhere in the namespace, because there is no unique path to scope to this assembly, so it will try to resolve from the parent project instead.

```razor
@using Blazor.IntersectionObserver
```

will throw

```diff
- The type or namespace name 'IntersectionObserver' does not exist in the namespace 'MyCompany.Blazor' 
- (are you missing an assembly reference?) [E:\Test\MyCompany.Blazor.MyProject\MyCompany.Blazor.MyProject.csproj]
```

## To Reproduce

1. Create new project where namespace includes `Blazor` in it
```powershell
PS> dotnet new blazorwasm -n MyCompany.Blazor.MyProject
```
2. Add reference to **BlazorIntesectionObserver**
```powershell
PS> cd MyCompany.Blazor.MyProject
PS> dotnet add package BlazorIntersectionObserver

```
3. Add using to _Imports.razor
```razor
@using Blazor.IntersectionObserver
```
4. Build the project
```powershell
PS> dotnet build
```

## Expected behavior
It should be possible to reference components in this library through unique namespace scope that is not commonly used eg.

```razor
@using Ljbc1994.Blazor.IntersectionObserver
```

## Additional context
https://github.com/dotnet/aspnetcore/issues/13090#issuecomment-807052901

**Comment by ljbc1994** [comment]
Thanks for raising this @Xeevis, that's a real pain.

When I have time, I'll have a look and resolve this.

**Comment by ljbc1994** [comment]
This has been updated and is available in version 3.0.0.

Thanks for raising let me know if you have any issues
